SpatialGDK/Source/SpatialGDKEditor/Private/SnapshotGenerator/SpatialGDKEditorSnapshotGenerator.cpp

```diff
@@ -166,6 +160,41 @@ bool CreateGlobalStateManager(Worker_SnapshotOutputStream* OutputStream)
 	return Worker_SnapshotOutputStream_GetState(OutputStream).stream_state == WORKER_STREAM_STATE_GOOD;
 }
 
+Worker_ComponentData CreateVirtualWorkerTranslatorData()
+{
+	Worker_ComponentData VirtualWorkerTranslatorData{};
+	VirtualWorkerTranslatorData.component_id = SpatialConstants::VIRTUAL_WORKER_TRANSLATION_COMPONENT_ID;
+	VirtualWorkerTranslatorData.schema_type = Schema_CreateComponentData();
+	return VirtualWorkerTranslatorData;
+}
+
+bool CreateVirtualWorkerTranslator(Worker_SnapshotOutputStream* OutputStream)
+{
+	Worker_Entity VirtualWorkerTranslator;
+	VirtualWorkerTranslator.entity_id = SpatialConstants::INITIAL_VIRTUAL_WORKER_TRANSLATOR_ENTITY_ID;
+
+	TArray<Worker_ComponentData> Components;
+
+	WriteAclMap ComponentWriteAcl;
+	ComponentWriteAcl.Add(SpatialConstants::POSITION_COMPONENT_ID, SpatialConstants::UnrealServerPermission);
+	ComponentWriteAcl.Add(SpatialConstants::METADATA_COMPONENT_ID, SpatialConstants::UnrealServerPermission);
+	ComponentWriteAcl.Add(SpatialConstants::PERSISTENCE_COMPONENT_ID, SpatialConstants::UnrealServerPermission);
+	ComponentWriteAcl.Add(SpatialConstants::ENTITY_ACL_COMPONENT_ID, SpatialConstants::UnrealServerPermission);
+	ComponentWriteAcl.Add(SpatialConstants::VIRTUAL_WORKER_TRANSLATION_COMPONENT_ID, SpatialConstants::UnrealServerPermission);
+
+	Components.Add(Position(Origin).CreatePositionData());
+	Components.Add(Metadata(TEXT("VirtualWorkerTranslator")).CreateMetadataData());
+	Components.Add(Persistence().CreatePersistenceData());
+	Components.Add(CreateVirtualWorkerTranslatorData());
+	Components.Add(EntityAcl(CreateReadACLForAlwaysRelevantEntities(), ComponentWriteAcl).CreateEntityAclData());
```

Comment:
This breaks the current authority intent implementation for the spatial debugger because it relies on the receiving the translation mapping via the GSM single-shot query. 

This results in this:

![image](https://user-images.githubusercontent.com/9222722/72094636-b9eb8480-330e-11ea-898f-34ddbb1c3823.png)

This would be fixed by https://improbableio.atlassian.net/browse/UNR-2565

Reply:
Still working on the debugger component. I won't merge this until we decide on the ordering.